### PR TITLE
also define $LD in buildopts for GATE

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

File: easybuild/easyblocks/g/gate.py [hunk 1]
```diff
@@ -92,8 +92,17 @@ def build_step(self):
...
+        # make sure right compilers and compiler options are used for building
+        opt_vals = {
+            'CC': os.getenv('CC'),
+            'CFLAGS': cflags,
+            'CXX': os.getenv('CXX'),
+            'CXXFLAGS': cxxflags,
+            # filemerger Makefile hardcodes $LD to g++, so make sure right compiler is used for linking
+            'LD': os.getenv('CXX'),
+        }
+        make_opts = 'CC="%(CC)s" CFLAGS="%(CFLAGS)s" CXX="%(CXX)s" CXXFLAGS="%(CXXFLAGS)s" LD="%(LD)s"' % opt_vals
```
Comment (wpoely86) [inline easybuild/easyblocks/g/gate.py]: this can be done nicer?

Comment (boegel) [inline easybuild/easyblocks/g/gate.py]: better now?

**Comment by wpoely86** [comment]
:+1:

**Comment by boegel** [comment]
testing with all existing `GATE` easyconfigs + the ones in https://github.com/hpcugent/easybuild-easyconfigs/pull/2623, works like a charm now, so going in

Thanks for the review @wpoely86!

**Pull request merged**